plugins/builtin/strategy/threshold/plugin/plugin.go

```diff
@@ -0,0 +1,296 @@
+		"actionType", config.actionType)
+
+	// Check if we have enough data points within bounds.
+	if !withinBounds(logger, eval.Metrics, config) {
+		logger.Trace("not enough data points within bounds")
+		return nil, nil
+	}
+
+	// Calculate new count.
+	logger.Trace("calculating new count")
+
+	var newCount int64
+	switch config.actionType {
+	case runConfigKeyDelta:
+		newCount = runDelta(count, config.actionValue)
+	case runConfigKeyPercentage:
+		newCount = runPercentage(count, config.actionValue)
+	case runConfigKeyValue:
+		newCount = runValue(count, config.actionValue)
+	}
+
+	logger = logger.With("new_count", newCount)
+
+	// Identify the direction of scaling, and exit early if none.
+	eval.Action.Direction = calculateDirection(count, newCount)
+	if eval.Action.Direction == sdk.ScaleDirectionNone {
+		return eval, nil
+	}
+
+	logger = logger.With("direction", eval.Action.Direction)
+	logger.Trace("calculated scaling strategy results")
+
+	eval.Action.Count = newCount
+	eval.Action.Reason = fmt.Sprintf("scaling %s because metric is within bounds", eval.Action.Direction)
+
+	return eval, nil
+}
+
+// parseConfig parses and validates the policy check config.
+func parseConfig(config map[string]string) (*thresholdPluginRunConfig, error) {
+	c := &thresholdPluginRunConfig{}
+
+	// Read and parse threshold bounds from check config.
+	upperStr := config[runConfigKeyUpperBound]
+	lowerStr := config[runConfigKeyLowerBound]
+	if upperStr == "" && lowerStr == "" {
+		return nil, fmt.Errorf("missing required field, must have either %q or %q", runConfigKeyLowerBound, runConfigKeyUpperBound)
+	}
+
+	upper, err := parseBound(runConfigKeyUpperBound, upperStr)
+	if err != nil {
+		return nil, err
+	}
+	c.upperBound = upper
+
+	lower, err := parseBound(runConfigKeyLowerBound, lowerStr)
+	if err != nil {
+		return nil, err
+	}
+	c.lowerBound = lower
+
+	// Read and parse within bounds trigger from check config.
+	triggerStr := config[runConfigKeyWithinBoundsTrigger]
+
+	trigger := defaultWithinBoundsTrigger
+	if triggerStr != "" {
+		t, err := strconv.ParseInt(triggerStr, 10, 0)
+		if err != nil {
+			return nil, fmt.Errorf("invalid value for %q: %v (%T)", runConfigKeyWithinBoundsTrigger, triggerStr, triggerStr)
+		}
+		trigger = int(t)
+	}
+	c.withinboundsTrigger = trigger
+
+	// Read and validate action type from check config.
+	deltaStr := config[runConfigKeyDelta]
+	percentageStr := config[runConfigKeyPercentage]
+	valueStr := config[runConfigKeyValue]
+
+	nonEmpty := 0
+	for _, s := range []string{deltaStr, percentageStr, valueStr} {
+		if s != "" {
+			nonEmpty++
+		}
+	}
+	if nonEmpty == 0 {
+		return nil, fmt.Errorf("missing required field, must have either %q, %q or %q", runConfigKeyDelta, runConfigKeyPercentage, runConfigKeyValue)
+	}
+	if nonEmpty != 1 {
+		return nil, fmt.Errorf("only one of %q, %q or %q must be provided", runConfigKeyDelta, runConfigKeyPercentage, runConfigKeyValue)
+	}
```

Comment:
I wonder if it is worth simplifying this into a single `!= 1` check rather than two checks to slightly change the error message grammar.

Reply:
Hum...not sure if I follow. `== 0` is for the case where no action type is provided. `!= 1` is for when more than one is provided. I guess it could be `> 1` for clarity.

But the error message in both cases would be different.